src/client.ts

```diff
@@ -615,46 +626,110 @@ export class VocdoniSDKClient {
+   * @returns {Promise<CensusProof>}
    */
-  private fetchProofForWallet(election: PublishedElection, wallet: Wallet | Signer): Promise<OffchainCensusProof> {
-    return wallet
-      .getAddress()
-      .then((address) => this.fetchProof(election.census.censusId, address, CensusProofType.ADDRESS));
+  private fetchProofForWallet(election: PublishedElection, wallet: Wallet | Signer): Promise<CensusProof> {
+    return wallet.getAddress().then((address) => this.fetchProof(election.census.censusId, address));
+  }
+
+  private async setAccountSIK(
+    election: PublishedElection,
+    sik: string,
+    censusProof: CensusProof,
+    wallet: Wallet | Signer
+  ): Promise<void> {
+    const address = await wallet.getAddress();
+    const calculatedSIK = await calcSik(address, sik);
+    const registerSIKTx = AccountCore.generateRegisterSIKTransaction(election.id, calculatedSIK, censusProof);
+    const chainId = await this.fetchChainId();
+    return AccountCore.signTransaction(registerSIKTx, chainId, wallet)
+      .then((signedTx) => ChainAPI.submitTx(this.url, signedTx))
+      .then((data) => this.waitForTransaction(data.hash));
+  }
+
+  /**
+   * Calculates ZK proof from given wallet.
+   *
+   * @param election
+   * @param wallet
+   * @returns {Promise<ZkProof>}
+   */
+  private async calcZKProofForWallet(election: PublishedElection, wallet: Wallet | Signer): Promise<ZkProof> {
+    const address = await wallet.getAddress();
+    const sik = await Signing.signRaw(new Uint8Array(Buffer.from(VOCDONI_SIK_PAYLOAD)), wallet);
+    const censusProof = await this.fetchProofForWallet(election, wallet);
+
+    await AccountAPI.info(this.url, address)
+      .then((account) => {
+        if (account.sik.length === 0) {
+          throw new Error();
+        }
+      })
+      .catch(() => this.setAccountSIK(election, sik, censusProof, wallet));
+
+    return ZkAPI.proof(this.url, address)
+      .then((zkProof) =>
+        prepareCircuitInputs(
+          election.id,
+          address,
+          '0',
+          sik,
+          censusProof.value,
+          censusProof.value,
+          zkProof.censusRoot,
+          zkProof.censusSiblings,
+          censusProof.root,
+          censusProof.siblings
+        )
+      )
+      .then((circuits) => this.generateZkProof(circuits));
   }
 
   /**
    * Creates an account with information.
    *
-   * @param {{account: Account, faucetPackage: string | null}} options Additional options,
+   * @param {{account: Account, faucetPackage: string | null, signedSikPayload: string | null}} options Additional options,
    * like extra information of the account, or the faucet package string.
    * @returns {Promise<AccountData>}
    */
-  async createAccountInfo(options: { account: Account; faucetPackage?: string }): Promise<AccountData> {
+  async createAccountInfo(options: {
+    account: Account;
+    faucetPackage?: string;
+    signedSikPayload?: string;
+  }): Promise<AccountData> {
```

Comment:
Be careful of setting defaults using objects. Since you're using an object, and you're not merging the given `options` with a defaults, when someone passes a settings object, if they miss any of the options, it won't be there.

This is also the case for createAccount. In general, if you want to use default params, it's more easier by not using objects (in this case, using three arguments for the function).